src/plan_expand_hypertable.c

```diff
@@ -139,6 +193,63 @@ ts_plan_expand_hypertable_valid_hypertable(Hypertable *ht, Query *parse, Index r
 	return true;
 }
 
+/**
+ * Get chunk oids from either restrict info or explicit chunk exclusion. Explicit chunk exclusion
+ * takes precedence.
+ */
+static List *
+get_chunk_oids(CollectQualCtx *ctx, PlannerInfo *root, RelOptInfo *rel, Hypertable *ht)
+{
+	List	   *chunk_oids = NIL;
+
+	if (ctx->chunk_exclusion_func == NULL)
+	{
```

Comment:
Can we move the blocks of this if-else clause into separate functions (especially the else part)?

Reply:
fixed